flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/OneInputStreamTaskTest.java

```diff
@@ -996,6 +1002,88 @@ public void testCanEmitBatchOfRecords() throws Exception {
         }
     }
 
+    @Test
+    public void testTaskSideOutputStatistics() throws Exception {
+        TaskMetricGroup taskMetricGroup =
+                UnregisteredMetricGroups.createUnregisteredTaskMetricGroup();
+
+        ResultPartitionWriter[] partitionWriters = new ResultPartitionWriter[3];
+        for (int i = 0; i < partitionWriters.length; ++i) {
+            partitionWriters[i] =
+                    new RecordOrEventCollectingResultPartitionWriter<>(
+                            new ArrayDeque<>(),
+                            new StreamElementSerializer<>(
+                                    BasicTypeInfo.INT_TYPE_INFO.createSerializer(
+                                            new ExecutionConfig())));
+            partitionWriters[i].setup();
+        }
+
+        try (StreamTaskMailboxTestHarness<Integer> testHarness =
+                new StreamTaskMailboxTestHarnessBuilder<>(
+                                OneInputStreamTask::new, BasicTypeInfo.INT_TYPE_INFO)
+                        .addInput(BasicTypeInfo.INT_TYPE_INFO)
+                        .addAdditionalOutput(partitionWriters)
+                        .setupOperatorChain(new OperatorID(), new OddEvenOperator())
+                        .chain(BasicTypeInfo.INT_TYPE_INFO.createSerializer(new ExecutionConfig()))
+                        .setOperatorFactory(SimpleOperatorFactory.of(new OddEvenOperator()))
+                        .addNonChainedOutputsCount(
+                                new OutputTag<>("odd", BasicTypeInfo.INT_TYPE_INFO), 2)
+                        .addNonChainedOutputsCount(1)
```

Comment:
Why we need to pass a tag for one but not for the other? 🤔

Reply:
When writing this test case, I felt a bit like just considering one of the two tags seemed sufficient. In this test case, no matter how the data is sent, only one `OutputTag` (`Odd` or `Even`) will actually be hit, which should be a dual relationship and will not affect the coverage of this test.

What I actually want to consider here is the scenario where both `RecordWriter Without Tag` and `RecordWriter With Tag` exist, which is also the actual topology of this case. For more details, please refer to the picture attached in the other comments.

Of course, if you think it is necessary to cover all tags here, I think it also makes sense.